irohad/main/irohad.cpp

```diff
@@ -272,7 +354,26 @@ int main(int argc, char *argv[]) {
 
     std::unique_ptr<iroha::ametsuchi::PostgresOptions> pg_opt;
     std::unique_ptr<iroha::ametsuchi::RocksDbOptions> rdb_opt;
-    if (config.database_config) {
+
+    // If legacy config precedence is disabled AND pg creds are specified in
+    // the environment variables, then apply those instead of parsing the
+    // config file's "pg_opt" property. The extra check with the flag ensures
+    // backwards compatibility.
+    if (auto db_config_env = getPostgresCredsFromEnv(config.database_config);
+        !FLAGS_legacy_config_precedence and db_config_env.has_value()) {
+      log->info("Parsing env vars for PG creds");
+      auto db_config = std::move(db_config_env).value();
```

Comment:
It is not necessarry to call move ctor.
```c++
auto const &db_config = db_config_env.value();

Reply:
@iceseer Done: b59d884557bde2e12975a528a3de3a2c23a6ca61